Event type error says mitmproxy has crashed without actually crashing

#### Problem Description
Some sort of event type error...? This happened when I was testing `modify_body` and `modify_headers` options (see reproduction steps) and it results in repeated error messages with really long tracebacks.

The error messages begin by saying that mitmproxy has crashed, but it doesn't actually crash.

```
[... lots more traceback before the below]
  File "/home/elespike/venvs/mitmproxy/lib/python3.9/site-packages/mitmproxy/proxy/layer.py", line 144, inhandle_event
    command = command_generator.send(send)
  File
"/home/elespike/venvs/mitmproxy/lib/python3.9/site-packages/mitmproxy/proxy/layers/http/__init__.py", line136, in _handle_event
    yield from self.client_state(event)
  File "/home/elespike/venvs/mitmproxy/lib/python3.9/site-packages/mitmproxy/proxy/utils.py", line 23, in
_check_event_type
    raise AssertionError(
AssertionError: Unexpected event type at HttpStream.state_done: Expected no events, got
RequestEndOfMessage(stream_id=1).
```

#### Steps to reproduce the behavior:
1. Start with default options
2. Set `modify_body` and `modify_headers` as follows:
```
modify_body:
  - :~u webhook.site ~q::<h1>hey</h1>

modify_headers:
  - :~u webhook.site ~s:content-type:text/html
```
3. Visit https://webhook.site

#### System Information
Mitmproxy: 7.0.4
Python:    3.9.2

--- a/mitmproxy/proxy/layers/websocket.py
+++ b/mitmproxy/proxy/layers/websocket.py
@@ -86,7 +86,6 @@ class WebsocketLayer(layer.Layer):
     def __init__(self, context: Context, flow: http.HTTPFlow):
         super().__init__(context)
         self.flow = flow
-        assert context.server.connected
 
     @expect(events.Start)
     def start(self, _) -> layer.CommandGenerator[None]:

